### ISSUE TITLE
Use ARIA labels for client search

### ISSUE THREAD

**Issue by klemay** [hypothesis/product-backlog#778]
Part of https://github.com/hypothesis/product-backlog/issues/753

## Requirements
- [ ] The form element needs `role="search"`
- [ ] The text input should have `type="search"` instead of `type="text"`
- [ ] The search button should have an `aria-label="search"`

**Comment by ajpeddakotla** [comment]
| User Engagement | Building Community | Level of Effort |
|---|---|---|
| 2 | 1 | 3 |

**Comment by ajpeddakotla** [comment]
Closed in favor of #1085
